File: KnightsTour.cpp
#include <iostream>;
using namespace std;

const int SizeOfBoard = 4;
const int necesseryMoves = (SizeOfBoard*SizeOfBoard) - 1;

bool CheckPath(int x,int y,int Size, bool visited[][SizeOfBoard],int moves)
{
	
	if ((x<0) || (x >= Size) || (y<0) || (y >= Size))
	{
		return false;
	}
	if (visited[x][y]==true)
	{
		
		return false;
	}
	if (moves == necesseryMoves)
	{
		cout << x << y << endl;
		visited[x][y] = true;
	
	}
	else
	{
		visited[x][y] = true;
		bool res = false;
		res = (res || (CheckPath(x + 2, y + 1, Size, visited, moves + 1)));
		if (res == false)
		{
			res = (res || (CheckPath(x + 2, y - 1, Size, visited, moves + 1)));
		}
		if (res == false){
			res = (res || (CheckPath(x - 2, y + 1, Size, visited, moves + 1)));
	}
		if (res == false)
		{
			res = (res || (CheckPath(x - 2, y - 1, Size, visited, moves + 1)));
	    }
		if (res == false){
			res = (res || (CheckPath(x + 1, y + 2, Size, visited, moves + 1)));
	}
		if (res == false){
			res = (res || (CheckPath(x + 1, y - 2, Size, visited, moves + 1)));
	}
		if (res == false){
			res = (res || (CheckPath(x - 1, y + 2, Size, visited, moves + 1)));
	}
		if (res == false){
			res = (res || (CheckPath(x - 1, y - 2, Size, visited, moves + 1)));
		}
		if (res==true)
		{
			cout << x <<"---"<< y << endl;
			
			
			return true;
		}
		else
		{
			visited[x][y] = false;
			return false;
		}
	}
}
int main()
{
	cout << "c++";
	bool visitedFeilds[SizeOfBoard][SizeOfBoard] = { false };


	if (CheckPath(0, 0, SizeOfBoard, visitedFeilds, 0) == true)
	{
		cout << "true";
	}
	else
	{
		cout << "false";
	}

	int x;
	cin >> x;
}

File: KnightsTour.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace exercise
{
    class KnightsTour
    
 {
        
        static void Main(string[] args)
        {
            int x =7;
            bool[,] visited = new bool[x, x];

            int necesseryMoves = (x * x) - 1;
            Console.WriteLine(Move(0, 0, 0, necesseryMoves, visited));

        }
        static bool Move(int x, int y, int moves, int necesseryMoves, bool[,] visited)
        {

            
            if ((x < 0) || (x >= visited.GetLength(0) || (y < 0) || (y >= visited.GetLength(1))))
            {
                return false;
            }
            
            if (visited[x, y] == true)
            {
                return false;
            }

            if (moves == necesseryMoves)
            {
                visited[x,y] = true;
                return true;
            }
            else
            {
                visited[x, y] = true;
                bool res = false;
                res = (res || (Move(x + 1, y + 2, moves + 1, necesseryMoves, visited)));
                res = (res || (Move(x + 2, y - 1, moves + 1, necesseryMoves, visited)));
                res = (res || (Move(x + 2, y + 1, moves + 1, necesseryMoves,visited)));
                res = (res || (Move(x + 2, y - 1, moves + 1, necesseryMoves, visited)));
                res = (res || (Move(x - 2, y + 1,  moves + 1, necesseryMoves, visited)));
                res = (res || (Move(x - 2, y - 1,  moves + 1, necesseryMoves, visited)));
                res = (res || (Move(x + 1, y - 2,  moves + 1, necesseryMoves, visited)));
                res = (res || (Move(x - 1, y + 2,  moves + 1, necesseryMoves, visited)));
                res = (res || (Move(x - 1, y - 2,  moves + 1, necesseryMoves, visited)));
                if (res == true)
                {
                    Console.WriteLine("x --> " + x);
                    Console.WriteLine("y --> " + y);
                    Console.WriteLine();
                    
                    return true;
                   
                }
                else
                {
                    visited[x,y] = false;
                    return false;
                }
            }
        }



    }
}
